clients/src/main/java/org/apache/kafka/common/security/oauthbearer/internals/unsecured/OAuthBearerUnsecuredValidatorCallbackHandler.java

```diff
@@ -170,33 +170,31 @@ private void handleCallback(OAuthBearerValidatorCallback callback) {
                 .throwExceptionIfFailed();
         OAuthBearerValidationUtils.validateTimeConsistency(unsecuredJwt).throwExceptionIfFailed();
         OAuthBearerValidationUtils.validateScope(unsecuredJwt, requiredScope).throwExceptionIfFailed();
-        log.info("Successfully validated token with principal {}: {}", unsecuredJwt.principalName(),
-                unsecuredJwt.claims().toString());
+        if (log.isInfoEnabled()) {
+            log.info("Successfully validated token with principal {}: {}", unsecuredJwt.principalName(), unsecuredJwt.claims());
+        }
         callback.token(unsecuredJwt);
     }
 
     private String principalClaimName() {
         String principalClaimNameValue = option(PRINCIPAL_CLAIM_NAME_OPTION);
-        String principalClaimName = principalClaimNameValue != null && !principalClaimNameValue.trim().isEmpty()
+        return principalClaimNameValue != null && !principalClaimNameValue.trim().isEmpty()
```

Comment:
How about adding a helper method to check both "null" and "empty"?

Reply:
Done.